src/OrchardCore.Modules/OrchardCore.Sitemaps/Handlers/CustomPathSitemapSourceUpdateHandler.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using OrchardCore.ContentManagement;
+using OrchardCore.Sitemaps.Models;
+using OrchardCore.Sitemaps.Services;
+
+namespace OrchardCore.Sitemaps.Handlers
+{
+    public class CustomPathSitemapSourceUpdateHandler : ISitemapSourceUpdateHandler
+    {
+        private readonly ISitemapManager _sitemapManager;
+
+        public CustomPathSitemapSourceUpdateHandler(ISitemapManager sitemapManager)
+        {
+            _sitemapManager = sitemapManager;
+        }
+
+        public async Task UpdateSitemapAsync(SitemapUpdateContext context)
+        {
+            var contentItem = context.UpdateObject as ContentItem;
```

Comment:
This won't work because there will never be a `ContentItem` that is related to this sitemap, because it is a custom path.

The function of this method is to clear the sitemape cache if a content item changes. (it's slightly not working see: https://github.com/OrchardCMS/OrchardCore/issues/8087 )

But with a custom path I don't think there will be a way to detect that the custom path has changed. So probably best to just remove this update handler